src/main/groovy/com/bmuschko/gradle/docker/utils/RegistryAuthLocator.groovy

```diff
@@ -0,0 +1,261 @@
+                }
+            }
+        }
+        return null
+    }
+
+    /**
+     * Checks 'credHelpers' section in the config json matching the given repository
+     * @param config config json object
+     * @param repository the name of the docker repository
+     * @return auth object if present or null otherwise
+     */
+    private AuthConfig authConfigUsingHelper(JsonNode config, String repository)  {
+        JsonNode credHelpers = config.get(HELPERS_SECTION)
+        if (credHelpers != null && credHelpers.size() > 0) {
+            JsonNode helperNode = credHelpers.get(repository)
+            if (helperNode != null && helperNode.isTextual()) {
+                String helper = helperNode.asText()
+                return runCredentialProvider(repository, helper)
+            }
+        }
+        log.debug('No helper found in the {} section', HELPERS_SECTION)
+        return null
+    }
+
+    /**
+     * Runs external credentials provider tool (e.g. docker-credential-gcloud)
+     * @param hostName the name of the docker repository to get auth for
+     * @param credHelper the suffix of the docker credential helper (e.g. gcloud)
+     * @return auth object if present or null otherwise
+     */
+    private AuthConfig runCredentialProvider(String hostName, String credHelper) {
+        String credentialHelperName = commandPathPrefix + credHelper
+
+        String data = runCommand(hostName, credentialHelperName)
+        JsonNode helperResponse = OBJECT_MAPPER.readTree(data)
+        log.debug('Credential helper provided auth config for: {}', hostName)
+
+        return new AuthConfig()
+            .withRegistryAddress(helperResponse.at(SERVER_URL).asText())
+            .withUsername(helperResponse.at(USER_NAME).asText())
+            .withPassword(helperResponse.at(SECRET).asText())
+    }
+
+    /**
+     * Checks 'credsStore' section in the config json matching the given repository
+     * @param config config json object
+     * @param repository the name of the docker repository
+     * @return auth object if present or null otherwise
+     */
+    private AuthConfig authConfigUsingStore(JsonNode config, String repository) {
+        JsonNode credsStoreNode = config.get(CREDS_STORE_SECTION)
+        if (credsStoreNode != null && !credsStoreNode.isMissingNode() &&
+                credsStoreNode.isTextual()) {
+            String credsStore = credsStoreNode.asText()
+            return runCredentialProvider(repository, credsStore)
+        }
+        log.debug('No helper found in the {} section', CREDS_STORE_SECTION)
+        return null
+    }
+
+    private static String runCommand(String hostName, String credentialHelperName) {
+        log.debug('Executing docker credential helper: {} to locate auth config for: {}',
+            credentialHelperName, hostName)
+        try {
+            ProcessBuilder pb = new ProcessBuilder(credentialHelperName, 'get')
+            Process process = pb.start()
```

Comment:
We could probably just use [built-in groovy](https://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/String.html#execute()) to do this in a more simple way.

Reply:
@cdancy Thanks for the suggestion, I am done with it